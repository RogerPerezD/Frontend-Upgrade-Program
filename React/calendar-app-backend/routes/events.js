const { Router } = require('express');
const { index, store, update, drop } = require('../controllers/EventsController');
const { validateJWT } = require('../middlewares/validate-jwt');
const router = Router();

// Hacer que todas las rutas usen el middleware para autenticar el jwt
router.use( validateJWT );

router.get('/',index);

router.post('/store',store);

router.put('/:id',update);

router.delete('/:id',drop);


module.exports = router;