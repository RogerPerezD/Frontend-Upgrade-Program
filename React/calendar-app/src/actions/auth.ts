import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch"
import { User, DispatchAuth } from '../reducers/authReducer';
import { types } from '../types/types';
import Swal from 'sweetalert2';

export const startLogin = ( email: string, password: string) =>{
    return async ( dispatch: DispatchAuth)=>{
        
        const resp = await fetchWithoutToken('auth', {email, password}, 'POST');
        const body = await resp.json();

        // console.log(body);
        if ( body.ok ) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime().toString());
        
            dispatch( 
                login({
                _id: body.uid,
                name: body.name
                }) 
            );
        }else{
            Swal.fire('Error', body.msg, 'error');
        }

    }
}

const login = ( user: User) =>{
    return {
        type: types.authLogin,
        payload: user
    }
}

export const startRegister = ( email: string, password: string, name: string) =>{
    return async ( dispatch: DispatchAuth )=>{
        
        const resp = await fetchWithoutToken('auth/new', {email, password, name}, 'POST');
        const body = await resp.json();

        // console.log(body);
        if ( body.ok ) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime().toString());
        
            dispatch( 
                register({
                _id: body.uid,
                name: body.name
                }) 
            );
        }else{
            Swal.fire('Error', body.msg, 'error');
        }

    }
}


const register = ( user: User) =>{
    return {
        type: types.authStartRegister,
        payload: user
    }
}

export const startChecking = ()=>{
    return async ( dispatch: DispatchAuth )=>{
        
        const resp = await fetchWithToken('auth/renew');
        const body = await resp.json();

        // console.log(body);
        if ( body.ok ) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime().toString());
        
            dispatch( 
                login({
                _id: body.uid,
                name: body.name
                }) 
            );
        }else{
            dispatch(checkingFinish());
        }

    }
}

const checkingFinish = ()=>{
    return {
        type: types.authCheckingFinish
    }
}

export const startLogout = ()=>{
    return (dispatch: DispatchAuth) =>{
        // localStorage.removeItem('token');
        // localStorage.removeItem('token-init-date');
        localStorage.clear();
        dispatch( logout() );
    }
}

const logout = ()=>{
    return {
        type: types.authLogout
    }
}