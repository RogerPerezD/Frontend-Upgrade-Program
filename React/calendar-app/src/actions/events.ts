// import { CustomEvent } from '../components/calendar/CalendarScreen';
import { EventAction, Event } from '../reducers/calendarReducer';
import { types } from '../types/types';

export const eventAddNew = ( event: Event ) =>{
    return {
        type: types.eventAddNew,
        payload: event
    }
}

export const eventSetActive = ( event: Event ) =>{
    return {
        type: types.eventSetActive,
        payload: event
    }
}

export const eventClearActiveEvent = ( ) =>{
    return {
        type: types.eventClearActiveEvent
    }
}

export const eventUpdated = ( event: Event ) =>{
    return {
        type: types.eventUpdated,
        payload: event
    }
}