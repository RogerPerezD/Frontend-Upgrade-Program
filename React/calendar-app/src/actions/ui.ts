import { types } from '../types/types';
import { uiAction } from '../reducers/uiReducer';

export const openModalAction = () =>{
    return{
        type: types.uiOpenModal
    }
}


export const closeModalAction = ():uiAction =>{
    return{
        type: types.uiCloseModal
    }
}