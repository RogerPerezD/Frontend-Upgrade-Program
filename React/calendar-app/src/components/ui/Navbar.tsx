import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';

export const Navbar = () => {
    const { user } = useSelector((state: RootState)=> state.auth);
    return (
        <div className='navbar navbar-dark bg-dark mb-4'>
            <span className='navbar-brand'>
                {user?.name}
            </span>

            <button className='btn btn-outline-danger'>
                <i className="fas fa-sign-out-alt"></i>
                <span> Salir</span>
            </button>

        </div>
    );
}
