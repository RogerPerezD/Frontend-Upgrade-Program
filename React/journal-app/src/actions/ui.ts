import { types } from '../types/types';
import { UIAction } from '../reducers/uiReducer';

export const setError = ( err: string ): UIAction=>{
    return {
        type: types.uiSetError,
        payload: err
    }
}

export const removeError = (): UIAction=>{
    return {
        type: types.uiRemoveError
    }
}