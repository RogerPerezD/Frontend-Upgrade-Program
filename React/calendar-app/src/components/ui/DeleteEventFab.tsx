import React from 'react'
import { useDispatch } from 'react-redux';
import { eventDeleted, eventClearActiveEvent } from '../../actions/events';

export const DeleteEventFab = () => {
    const dispatch = useDispatch();
    const handleDeleteEvent = () =>{
        dispatch( eventDeleted() );
        // dispatch( eventClearActiveEvent() );
    }
    return (
        <button
            className='btn btn-danger fab-danger'
            onClick={ handleDeleteEvent }
        >
            <i className='fas fa-trash'></i>
            <span>Delete Event</span>
        </button>
    )
}
