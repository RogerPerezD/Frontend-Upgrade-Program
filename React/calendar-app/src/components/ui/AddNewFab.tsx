import React from 'react'
import { useDispatch } from 'react-redux';
import { openModalAction } from '../../actions/ui';

export const AddNewFab = () => {
    const dispatch = useDispatch();
    const handleAddEvent = () =>{
        dispatch( openModalAction() );
    }
    return (
        <button
            className='btn btn-primary fab'
            onClick={ handleAddEvent }
        >
            <i className='fas fa-plus'></i>
        </button>
    );
}
