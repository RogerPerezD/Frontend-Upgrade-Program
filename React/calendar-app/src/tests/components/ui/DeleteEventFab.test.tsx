import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { RootState } from '../../../store/store';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';
// jest.mock('../../../actions/events', ()=>{
//     return {
//         eventStartDelete: jest.fn()
//     }
// });
jest.mock('../../../actions/events');
// Our mock of add is now fully typed
const mockDelete = eventStartDelete as jest.MockedFunction<typeof eventStartDelete>;

const middlewares = [ thunk ];
const mockStore = configureStore<RootState>(middlewares);
const initState = {} as RootState;
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
        <MemoryRouter>
            <Provider store = { store }>
                <DeleteEventFab/>
            </Provider>
        </MemoryRouter>
    );
describe('test in DeleteEventFab', () => { 

    test('should show the component successfully', () => { 
        expect( wrapper ).toMatchSnapshot();
    });

    test('should call eventStartDelete when click it', () => { 

        
        wrapper.find('button').simulate('click');

        expect( store.dispatch ).toBeCalled();
        expect(mockDelete).toHaveBeenCalled();
    })
})