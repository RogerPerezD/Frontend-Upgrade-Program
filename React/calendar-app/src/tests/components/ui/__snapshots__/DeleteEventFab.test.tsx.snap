// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in DeleteEventFab should show the component successfully 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Delete Event
  </span>
</button>
`;
